src/pages/api/gamer/[platform]/[username].ts

```diff
@@ -0,0 +1,122 @@
+            name: teammate.helping_player,
+            platform: teammate.helping_player_platform
+        };
+        return teammate;
+    });
+}
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+    let {view, username, platform } = req.query;
+    let queriedViews = (view as string).split(',');
+    let defaultViews = ['player_stat_summary'];
+    let views = {
+        'player_stat_summary': {
+            query: {
+                username: username,
+                platform: platform
+            }
+        },
+        'teammate_analysis': {
+            query: {
+                shooting_player: req.query.username,
+                shooting_player_platform: req.query.platform
+            }
+        },
+        'gamer_stats_graded': {
+            query: {
+                username: req.query.username,
+                platform: req.query.platform
+            }
+        },
+        'gamers': {
+            query: {
+                username: req.query.username,
+                platform: req.query.platform
+            }
+        },
+        'time_of_day_analysis': {
+            query: {
+                timezone: 'America/Los_Angeles',
+                username: req.query.username,
+                platform: req.query.platform,
+                cutoff: "10"
+            }
+        },
+        'day_of_week_analysis': {
+            query: {
+                timezone: 'America/Los_Angeles',
+                username: req.query.username,
+                platform: req.query.platform,
+                cutoff: "10"
+            }
+        }
+    };
+    let gamerQueryData = await queryGamers({username: username, platform: platform});
+    let gamerData =gamerQueryData[0]
+    if(gamerData){
+        let promises = Object.keys(views).filter((v)=> defaultViews.includes(v) || queriedViews.includes(v)).map(async (key: string) => await queryView(key, views[key].query));
+        Bluebird.all(promises).then(async (arrData) => {
+            Object.keys(views).filter((v)=> defaultViews.includes(v) || queriedViews.includes(v)).forEach((key: string, index: number) => views[key].data = arrData[index]);
+            let gamer = sanitizeGamer(views['player_stat_summary']['data'][0]) || {};
+            let gamerPromise = Bluebird.resolve(gamerData);
+            let viewData = views[view as string]['data'];
+            if(view as string === 'teammate_analysis'){
+                let newViewData = {};
+                newViewData['teammates']= sanitizeTeammates(viewData);
+                newViewData['titleKeys'] =  Object.keys(viewData[0]).filter((key) => !FILTER_KEYS.includes(key)).map(columnToDisplayName);
+                viewData = newViewData;
+            }
+            if (!gamerData.needs_update) {
+                gamerData.needs_update = true;
+                gamerPromise = updateGamer({username: gamerData.username, platform: gamerData.platform}, gamerData);
+            }
+            await gamerPromise;
```

Comment:
Can this simply go inside of the if statement?

like if the gamerPromise was removed and we just something like:

```
if (!gamerData.needs_update) {
    gamerData.needs_update = true;
    await updateGamer({username: gamerData.username, platform: gamerData.platform}, gamerData);
}
```
and completely nixing the Bluebird.resolve(gamerData); on line 86

Would that work the same? or do we have to have it on the lower level outside of the if statement?

Reply:
I don't know. I made it the promise way because I wasn't sure if Async/Await would handle a break in flow like that. It must work though since I tried it.